### DISCUSSION TITLE
Filters not visible wiith RefinableStringxxx associated to SP User Profile Property using Term Store

### DISCUSSION THREAD

**Discussion by Jrstew2016** [microsoft-search/pnp-modern-search#3318] [Q&A]
I'm having issues getting my filters to work for custom SharePoint user profile properties that are using a term set. I've tried associating the Refiner to the owc_taxid_userprofilecrawled property and it won't display. 
I got it to work partially once,  but it displayed only one value from the term set then several GUIDs which happened to all me my profile but different guid id's. 
I've created a custom managed property as well and associated with the crawled property too. So what am I missing here?

**Comment by kasperbolarsen** [comment]
As I see it there are two things you can try:
Try mapping a new RefinableString to ows_fieldname

Try to map another RefinableString to ows_fieldname AND remember to set the index to peopleidx in the advanced settings in the mapping dialog

The last option has worked for me for years 😁

**Reply by Jrstew2016** [reply]
When I change a RefinableString index to peopleidx it defaults back to default. 
I did manage to get crawled properties with OWS_propertyname to filter correctly. But I can't get the ones with ows_taxid_propertyname to work. See screenshots.
![filterpeopledirectory1](https://github.com/microsoft-search/pnp-modern-search/assets/149169590/d42e6d8a-eea9-49e7-9dca-2d03ca8a34b0)
![filterpeopledirectory2](https://github.com/microsoft-search/pnp-modern-search/assets/149169590/07d2d719-baae-40a5-8bcb-84006a3e7377)


 I've tried following this https://microsoft-search.github.io/pnp-modern-search/v3/search-parts/search-results/#other-properties but I'm kind of lost at this part: "When you enable this flag, you can still use the original property value for whatever reason by using the following property format in your Handlebars template: <propertyName> + TaxId (ex: owstaxidmetadataalltagsinfo will contain the translated terms and owstaxidmetadataalltagsinfoTaxId) will contain the original value of the managed property)."I know it's for version 3 so I would think it'd either be the same for V4 which I'm using or an enhancement already baked in now. Either way, I'm stuck!

**Reply by kasperbolarsen** [reply]
As I recall it,  you should only use the OWS_propertyname for Refiner as it contains the text value of the term. 
ows_taxid_propertyname contains the guid+ text  so it is not ideal for display purposes

**Reply by Jrstew2016** [reply]
Yea I realize that now. Whats odd is MSFTs logic on creating the ows_ items. List or site columns get the ows_name. While custom properties created in the SP user profiles gets the ows_taxid_name the crawled property and in most cases gets owstaxidname as the managed property. So I found  if you just use the PEOPLE:crawled property in the refinable managed property, it will display correct on the filter. But only values that have been uses in someones profile from that term set or group. 
I'd like to know to figure out how to use the custom filter feature extensibility feature but I'm no developer so it's tough for me.

Oh, side note on this. After all the work of creating those custom properties using the term store. The bulk Import API for SPO doesnt support taxonomic properties. I missed that part mainly because I just now got to the point of importing properties! Hah. 

But thanks for your help and being responsive!

**Reply by whitehotaru** [reply]
Hi @Jrstew2016! Did you try to enable localization in your configuration and map the property with the “Auto” prefix? Instead of writing “RefinableString130” try “AutoRefinableString130”. This works well for non people taxonomies. I didn't try it with people search yet, though.
